packages/emoji-blast-site/src/components/hero/hero.tsx

```diff
@@ -7,10 +7,29 @@ import { Anchor } from "../../components/anchor/anchor";
 import { Button } from "../button/button";
 import { Title } from "../title/title";
 import * as styles from "./styles";
+import { useCallback } from "react";
 
 export function Hero() {
 	useKonamiEmojiBlast();
 
+	const onExplosion = useCallback<React.MouseEventHandler<HTMLButtonElement>>(
+		(e) => {
```

Comment:
The `Button` component is pretty straightforward and doesn't care about referential equality of its props properties. And `Hero` isn't going to re-render very often. I think adding `useCallback` is a bit of a premature optimization here.

I don't particularly mind having this be a new function vs. an inline one, but let's drop the `useCallback`. I.e.:

```suggestion
	const onExplosion = (e) => {
```


(that, but with types - however you prefer to declare it as a mouse event handler)

https://stackoverflow.com/questions/64134566/should-we-use-usecallback-in-every-function-handler-in-react-function-components

Reply:
Yeah, I kinda forgot that React doesn't actually attach listeners to the DOM elements for VNodes, like Preact and Solid would do, so swapping is so cheap that this optimisation doesn't make much sense indeed. I removed `useCallback`, but kept it as as a separate function with type definition so that that the JSX template below is easy to read at a glance 👍